openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -1,14 +1,37 @@
 from rest_framework import serializers
+from django.conf import settings
 from django.contrib.auth.models import User
 from openedx.core.djangoapps.user_api.accounts import NAME_MIN_LENGTH
 from openedx.core.djangoapps.user_api.serializers import ReadOnlyFieldsSerializerMixin
 
-from student.models import UserProfile
+from student.models import UserProfile, LanguageProficiency
 from .helpers import get_profile_image_url_for_user, PROFILE_IMAGE_SIZES_MAP
 
 PROFILE_IMAGE_KEY_PREFIX = 'image_url'
 
 
+class LanguageProficiencySerializer(serializers.ModelSerializer):
+    """
+    Class that serializes the LanguageProficiency model for account
+    information.
+    """
+    class Meta:
+        model = LanguageProficiency
+        fields = ("code",)
+
+    def get_identity(self, data):
```

Comment:
I'll add an explicit test case, though nothing worked consistently without this method being overridden.